src/engine/QueryPlanner.cpp

```diff
@@ -865,44 +862,43 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::getDistinctRow(
       }
     }
     const std::vector<size_t>& resultSortedOn =
-        parent._qet->getRootOperation()->getResultSortedOn();
+        i._qet->getRootOperation()->getResultSortedOn();
     // check if the current result is sorted on all columns of the distinct
     // with the order of the sorting
     bool isSorted = resultSortedOn.size() >= keepIndices.size();
-    for (size_t i = 0; isSorted && i < keepIndices.size(); i++) {
-      isSorted = isSorted && resultSortedOn[i] == keepIndices[i];
+    for (size_t ii = 0; isSorted && ii < keepIndices.size(); ii++) {
+      isSorted = isSorted && resultSortedOn[ii] == keepIndices[ii];
     }
     if (isSorted) {
-      auto distinct =
-          std::make_shared<Distinct>(_qec, parent._qet, keepIndices);
+      auto distinct = std::make_shared<Distinct>(_qec, i._qet, keepIndices);
       distinctPlan._qet->setOperation(QueryExecutionTree::DISTINCT, distinct);
       distinctPlan._qet->setVariableColumns(distinct->getVariableColumns());
-      distinctPlan._qet->setContextVars(parent._qet->getContextVars());
+      distinctPlan._qet->setContextVars(i._qet->getContextVars());
     } else {
       if (keepIndices.size() == 1) {
         auto tree = std::make_shared<QueryExecutionTree>(_qec);
-        auto sort = std::make_shared<Sort>(_qec, parent._qet, keepIndices[0]);
-        tree->setVariableColumns(parent._qet->getVariableColumns());
+        auto sort = std::make_shared<Sort>(_qec, i._qet, keepIndices[0]);
+        tree->setVariableColumns(i._qet->getVariableColumns());
         tree->setOperation(QueryExecutionTree::SORT, sort);
-        tree->setContextVars(parent._qet->getContextVars());
+        tree->setContextVars(i._qet->getContextVars());
         auto distinct = std::make_shared<Distinct>(_qec, tree, keepIndices);
         distinctPlan._qet->setOperation(QueryExecutionTree::DISTINCT, distinct);
         distinctPlan._qet->setVariableColumns(distinct->getVariableColumns());
-        distinctPlan._qet->setContextVars(parent._qet->getContextVars());
+        distinctPlan._qet->setContextVars(i._qet->getContextVars());
       } else {
         auto tree = std::make_shared<QueryExecutionTree>(_qec);
         vector<pair<size_t, bool>> obCols;
-        for (auto& i : keepIndices) {
-          obCols.emplace_back(std::make_pair(i, false));
+        for (auto& ii : keepIndices) {
+          obCols.emplace_back(std::make_pair(ii, false));
```

Comment:
not `ii`,  or `k`. 
Also, in general I think you should not introduce too many new refactorings in places unrelated to the actual implementation. They slow down the merging process and possibly lead to merge conflicts with other PRs.

Please start separate PRs for such refactorings, unless they are really small and closely connected to your actual changes from the PR.